### PR TITLE
Add Project Owner responsibilities

### PR DESCRIPTION
**Pull request by dfarrell07**
See commit messages for details.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

**Review by nyechiel — APPROVED** [review]

**Comment by dfarrell07** [comment]
I made non-trivial changes to also define how Committers and Members are removed, so re-requested review @nyechiel.

**Review by nyechiel — APPROVED** [review]
LGTM, but I'd like @mangelajo @Oats87 @tpantelis @skitt to review as well

**Review by tpantelis — APPROVED** [review]

**Review by mangelajo — APPROVED** [review]
Looks ok to me but please make sure that all code owners are ok with L188 before merging, even if we don't have intentions to do anything like that at this time, I guess it's something we need to define as part of the governance.

**Review by skitt — APPROVED** [review]
PO removal seems a bit iffy to me, but I’m not sure there’s much more we can do. We’d need an oversight committee of some sort (like the TSC in other groups of projects), or perhaps company participation rules (to guarantee for example that we don’t end up with a dominant company’s employees voting to throw out a PO from another company).

**Comment by dfarrell07** [comment]
> PO removal seems a bit iffy to me, but I’m not sure there’s much more we can do. We’d need an oversight committee of some sort (like the TSC in other groups of projects), or perhaps company participation rules (to guarantee for example that we don’t end up with a dominant company’s employees voting to throw out a PO from another company).

Right, and we've help build systems like that in the past for OpenDaylight and other Linux Foundation Networking projects. I agree we should continue improving, especially as we mature (ie go for CNCF Incubation). For now I was just trying to get us started with something simple.

**Comment by dfarrell07** [comment]
Here's the system we helped build for OpenDaylight, for example: https://wiki.opendaylight.org/display/ODL/TSC+Election+Process

I think it's very robust and fair, but complex.

**Comment by Oats87** [comment]
@dfarrell07 et all: The PO removal process seems too open to "hostile takeover" to me, especially after seeing what happened with Freenode recently, I am very wary of such a process being implemented that doesn't have more barriers to prevent POs from a majority company performing a hostile takeover.

I'd be more open to a system of votes where PO votes that come from a single company would only count as one vote, but this proves to be difficult as there are only two companies on the project at this point that have POs.

File: src/content/community/contributor-roles/_index.en.md
Comment (Oats87) [inline src/content/community/contributor-roles/_index.en.md]: This seems a bit too open to hostile takeover possibilities to me.

**Review by Oats87 — CHANGES_REQUESTED** [review]

File: src/content/community/contributor-roles/_index.en.md
Comment (dfarrell07) [reply src/content/community/contributor-roles/_index.en.md]: @Oats87 With our current PO situation, I don't think there can be a system in which a PO can be removed and hostile takeovers are prevented. If it were "by vote of all POs other than the one being removed", it'd still be the same situation.

We'd need to do something more like a TSC I think, and have Represented Groups from stakeholders that aren't Red Hat/Rancher. Open to suggestions. I'll be thinking about it.

Comment (dfarrell07) [reply src/content/community/contributor-roles/_index.en.md]: We could just say "POs can't be removed except for Code of Conduct violations" for now, and build a TSC when we go for CNCF Incubation.

Comment (dfarrell07) [reply src/content/community/contributor-roles/_index.en.md]: ODL is currently using only a Committer at Large (CaL) Represented Group (RG) with a company cap of 49%. That is quite simple, good for smaller communities, but also provides protection from takeovers. 

https://wiki.opendaylight.org/display/ODL/TSC+Election+Process#TSCElectionProcess-TSCSeatAllocationsByRepresentedGroupforthe2021elections

Comment (dfarrell07) [reply src/content/community/contributor-roles/_index.en.md]: I know it's complex, but tl;dr that using that system, if people from 3 different companies ran, each company would be guaranteed at least one seat and we would be protected from takeovers.

Comment (mangelajo) [reply src/content/community/contributor-roles/_index.en.md]: The company cap makes sense. Ditching anybody of the project or doing takeovers is far from our intentions, would be a terrible start for a project ;-]

**Review by mangelajo — CHANGES_REQUESTED** [review]
let's resolve @Oats87 concerns before merging.

File: src/content/community/contributor-roles/_index.en.md
Comment (dfarrell07) [reply src/content/community/contributor-roles/_index.en.md]: > The company cap makes sense. 

To be clear, the company cap can't really exist without the rest of the system. Most of the complexity of the system comes from implementing the company cap while satisfying a number of other [important principles](https://wiki.opendaylight.org/display/ODL/TSC+Election+Process#TSCElectionProcess-Principles).

> would be a terrible start for a project

Right, and even if one imagines we have no individual ethics and are corporate bots it would be absurd for an open source company to willfully take the reputational damage from crushing a budding open source project. :sweat_smile:

Comment (dfarrell07) [reply src/content/community/contributor-roles/_index.en.md]: > ODL is currently using only a Committer at Large (CaL) Represented Group (RG) with a company cap of 49%. That is quite simple, good for smaller communities, but also provides protection from takeovers.

My intention was to propose something like this as a simple TSC-based solution, *once we make the move to a TSC*. I think there are no Sandbox projects with TSCs now, and I doubt there are many Incubation projects even.

For now I think we can just freeze removals except for CoC violations, and add some wording that we must change to adopt a TSC-based system by a year-or-so out and if not we default to ODL's system with only CaL RG and 49% company cap as described above.

Comment (skitt) [reply src/content/community/contributor-roles/_index.en.md]: Does a TSC make sense in a single-project context? Who would we expect to be in the TSC, and how would we ensure that it wasn’t just the project owners?

Comment (mangelajo) [reply src/content/community/contributor-roles/_index.en.md]: what's a TSC? :)

Comment (dfarrell07) [reply src/content/community/contributor-roles/_index.en.md]: > Does a TSC make sense in a single-project context?

Maybe it's better just to scope/call it "electing/removing project owners".

> Who would we expect to be in the TSC, and how would we ensure that it wasn’t just the project owners?

I imagined that by the time we go for Incubation, we would have a fairly substantial user community from which we could elect people. I'd hope that the current POs would run and be elected, in addition to new POs from other ~companies.

> what's a TSC? :)

Technical Steering Committee, elected project leadership.

Comment (Oats87) [reply src/content/community/contributor-roles/_index.en.md]: I think a current solution of freezing removals except for CoC violations makes sense, although I would say a TSC is going to require additional manpower to run; if you think that there is enough bandwidth in order to run such a committee then that would be OK, but otherwise, the ODL system with a CaL RG + 49% company cap is "simplest" past the CoC violation removal.

Comment (dfarrell07) [reply src/content/community/contributor-roles/_index.en.md]: @Oats87 Cool, yes sounds like a good path to me. I saw this a bit late to make changes to the PR before the meeting in 4m, but I'll send an update with that wording after. :+1: :pray:

Comment (dfarrell07) [reply src/content/community/contributor-roles/_index.en.md]: Okay, I sent updated wording (split into second commit). Re-requesting review from everyone.

File: src/content/community/contributor-roles/_index.en.md
Comment (skitt) [inline src/content/community/contributor-roles/_index.en.md]: This is like a sign saying “back in five minutes” on a shop. If we’re going to specify a calendar, we need to have actual dates here.

**Review by skitt — CHANGES_REQUESTED** [review]

File: src/content/community/contributor-roles/_index.en.md
Comment (mangelajo) [inline src/content/community/contributor-roles/_index.en.md:19]: The ** ^ is not closed, please see the render:

https://deploy-preview-520--elated-bell-2913d9.netlify.app/community/contributor-roles/

File: src/content/community/contributor-roles/_index.en.md
Comment (dfarrell07) [reply src/content/community/contributor-roles/_index.en.md]: Maybe one day we'll be able to run `git blame` against physical signs, but yes I see the point and made it explicit. :wink: :pray:

File: src/content/community/contributor-roles/_index.en.md
Comment (dfarrell07) [reply src/content/community/contributor-roles/_index.en.md:19]: Good catch, IDK how I missed that one. Thanks for the renders by the way, they are so cool/helpful!

**Review by nyechiel — COMMENTED** [review]
LGTM, @Oats87 @mangelajo @tpantelis @skitt can you review as well, please?

**Review by skitt — APPROVED** [review]

**Review by mangelajo — APPROVED** [review]

**Review by Oats87 — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Pull request merged**
